app/containers/Forms/ProductionFields.tsx

```diff
@@ -1,127 +1,224 @@
-                quantity: Number(
-                  (e.nativeEvent.target as HTMLInputElement).value
-                )
-              });
-            }}
-          />
-        </Form.Group>
-      </Col>
-      <Col xs={12} md={4}>
-        <Form.Group controlId="id.associatedEmissions">
-          <Form.Label>Associated Emissions (tCO2e)</Form.Label>
-          <Form.Control
-            type="number"
-            defaultValue={formData.associatedEmissions}
-            onChange={e => {
-              onChange({
-                ...formData,
-                associatedEmissions: Number(
-                  (e.nativeEvent.target as HTMLInputElement).value
-                )
-              });
-            }}
-          />
-        </Form.Group>
-      </Col>
-      <Col xs={12} md={12}>
-        <Form.Group controlId="id.comments">
-          <Form.Label>Comments</Form.Label>
-          <Form.Control
-            as="textarea"
-            type="string"
-            value={formData.comments}
-            onChange={e => {
-              onChange({
-                ...formData,
-                comments: (e.nativeEvent.target as HTMLInputElement).value
-              });
-            }}
-          />
-        </Form.Group>
-      </Col>
+            <registry.fields.StringField
+              disabled
+              readonly
+              required={false}
+              schema={schema.properties.productUnits as JSONSchema6}
+              uiSchema={uiSchema.productUnits}
+              formData={formData.productUnits}
+              autofocus={autofocus}
+              idSchema={idSchema}
+              registry={registry}
+              errorSchema={errorSchema}
+              formContext={formContext}
+              name="units"
+              onChange={() => {
+                console.error("don't change me");
+              }}
+            />
+          </FieldTemplate>
+        </>
+      )}
+      {getAdditionalDataSchema() && (
+        <registry.fields.ObjectField
+          required
+          schema={getAdditionalDataSchema().schema}
+          uiSchema={getAdditionalDataSchema().uiSchema}
+          formData={formData.additionalData}
+          autofocus={autofocus}
+          idSchema={idSchema}
+          registry={registry}
+          errorSchema={errorSchema}
+          formContext={formContext}
+          disabled={disabled}
+          readonly={readonly}
+          name={null}
+          onChange={handleAdditionalFieldsChange}
+        />
+      )}
     </>
   );
 };
 
-export default createFragmentContainer(ProductionFields, {
+export default createFragmentContainer(ProductionFieldsComponent, {
   query: graphql`
     fragment ProductionFields_query on Query {
-      allProducts {
+      allProducts(condition: {state: "active"}) {
         edges {
           node {
             name
             units
+            productFormByProductFormId {
+              productFormSchema
+            }
           }
         }
       }
```

Comment:
How big is the `productFormSchema`? Should we be fetching `n` records for first render?

Reply:
It's an object with a dozen properties (each with a type, title, min/max values, etc) at most. 
I'm sure this can be improved but I don't think it belongs in this PR